src/shell/dotnet/Shell/Utilities/ResourceReader.cs

```diff
@@ -0,0 +1,44 @@
+﻿/*
+ * Morgan Stanley makes this available to you under the Apache License,
+ * Version 2.0 (the "License"). You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership. Unless required by applicable law or agreed
+ * to in writing, software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Shell.Utilities
+{
+    public static class ResourceReader
+    {
+        public static string ReadResource(string name) 
+        {
+            var assembly = Assembly.GetExecutingAssembly();
+            string resourcePath = name;
+
+            var stream = assembly.GetManifestResourceStream(resourcePath);
+
+            if (stream == null)
+            {
+                throw new InvalidOperationException("Resource not found");
+            }
+            using (var reader = new StreamReader(stream))
```

Comment:
It's nicer to make use of the new using declaration instead of a using block when the block ends at the end of the parent scope.
e.g. using var reader = new StreamReader(stream)